sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -408,14 +408,19 @@ message WindowingStrategy {
   // (Required) The coder for the windows of this PCollection.
   string window_coder_id = 3;
 
+  // (Required) Whether or not the window fn assigns inputs to exactly one window
+  //
+  // This knowledge is required for some optimizations
+  bool assigns_to_one_window = 4;
+
   // (Required) The trigger to use when grouping this PCollection.
-  Trigger trigger = 4;
+  Trigger trigger = 5;
```

Comment:
You'll note that the ITs failed because of reindexing the fields :-)

We aren't at the point where we need stability here, but leaving them alone will save you a worker dance.